File: python/p016.py
import math

def compute(BASE, EXPONENT):
	product = str(int(math.pow(BASE, EXPONENT)))
	sum = 0
	for i in map(int, product):
		sum += i
	return sum

if __name__ == "__main__":
	print(compute(2, 1000))

File: cpp/p016.cpp
#include <iostream>
#include <cmath>
#include <string>

int compute(const int BASE, const int EXPONENT) {
	std::string product = std::to_string(pow(BASE, EXPONENT));
	int sum = 0;
	for (char &num : product) {
		if (num != '.' && num != '0')
			sum += num - '0';
	}
	return sum;
}

int main() {
	std::cout << compute(2, 1000) << std::endl;
	return 0;
}
